R/pkgapi.R

```diff
@@ -37,10 +51,7 @@ pkgapi <- R6::R6Class(
       ## NOTE: We could use a different method here rather than
       ## overloading handle, as to add plain plumber endpoints.
       assert_is(endpoint, "pkgapi_endpoint")
-      endpoint <- plumber::PlumberEndpoint$new(
-        endpoint$method, endpoint$path, endpoint$plumber, private$envir,
-        serializer = pkgapi_serialize_pass)
-      super$handle(endpoint = endpoint)
+      super$handle(endpoint = endpoint$create(private$envir, private$validate))
```

Comment:
This looks kind of weird to me - but perhaps I am missing something.

This is called with an endpoint but we use that to create a new one to set the serializer on it? We could achieve that by having the `pkgapi_endpoint` ctor set the serializer? I see it is doing something with passing around environments too but I'm not sure what that achieves.

Reply:
There is some unhelpful overloading of names here. This takes a `pkgapi_endpoint` and from it creates a `plumber::PlumberEndpoint` object, passing it through to the `plumber::plumber` object's `$handle` method.  The environment thing is frankly a mystery and the behaviour here is the same as plumber's (there's a comment about this in the pkgapi initialize method...)

Reply:
oh yeah - a `pkgapi_endpoint` isn't a `PlumberEndpoint`. That makes sense - thanks